dissect/target/helpers/scrape.py

```diff
@@ -53,13 +56,44 @@ def find_needles(
     offset = fh.tell() if start is None else start
     current_block = b""
 
+    # MappingStream gap skipping logic
+    is_mapping_stream = isinstance(fh, MappingStream)
+    if is_mapping_stream:  # We need to seek to skip over empty regions
+        if not lock_seek:
+            raise ValueError("lock_seek must be True when using a MappingStream")
+        if not (runs := fh._runs):
+            return
+        run_idx = 0
+
     while offset < end if end is not None else True:
+        # Special case for MappingStream to skip gaps
+        if is_mapping_stream:
+            # Find the next run that covers the current offset
+            while not (runs[run_idx][0] <= offset < runs[run_idx][0] + runs[run_idx][1]):
+                if offset < runs[run_idx][0]:  # There is a gap, jump
+                    current_block = b""  # No overlap when jumping to a new run
```

Comment:
This means we won´t  detect non-contiguous needles, i.e. spread over multiple mappings.